src/sc/sidechainTxsCommitmentBuilder.h

```diff
@@ -0,0 +1,62 @@
+#ifndef SIDECHAIN_TX_COMMITMENT_BUILDER
+#define SIDECHAIN_TX_COMMITMENT_BUILDER
+
+#include <map>
+#include <vector>
+#include <set>
+
+#include <zendoo/zendoo_mc.h>
+#include <uint256.h>
+
+class CTransaction;
+class CScCertificate;
+
+class SidechainTxsCommitmentBuilder
+{
+public:
+    void add(const CTransaction& tx);
+    void add(const CScCertificate& cert);
+    uint256 getCommitment();
+
+private:
+    // the final model will have a fixed-height merkle tree for 'SCTxsCommitment'.
+    // That will also imply having a limit per block to the number of crosschain outputs per SC
+    // and also to the number of SC
+    // -----------------------------
+
+    // Key:   the side chain ID
+    // Value: the array of fields into which forward transfers are mapped.
+    //        Tx are ordered as they are included in the block
+    std::map<uint256, std::vector<field_t*> > mScMerkleTreeLeavesFt;
+
+    // The same for BTR (to be implemented)
+    std::map<uint256, std::vector<field_t*> > mScMerkleTreeLeavesBtr;
+
+    // Key:   the side chain ID
+    // Value: the field into with the certificate related to this scid is mapped
+    std::map<uint256, field_t* > mScCerts;
+
+    // the catalog of scid, resulting after having collected all above contributions
+    std::set<uint256> sScIds;
+
+    class EmptyField
```

Comment:
It's not necessary true that an empty leaf/node, corresponds to "all zeros" (also looking at the implementation in the .cpp). I think that this class must depend on the height of the node for which we are computing the empty field (i.e. an empty leaf will have some default value (not necessarily all zeros), an empty node at level 1 will be the hash of two default leaves, ..., an empty node at level i will be the hash of two empty nodes at level i-1).
Of course all this values can be precomputed and used when necessary (I can easily provide a function in zendoo-mc-cryptolib)